src/app/data-planes/data/stats.ts

```diff
@@ -54,39 +99,57 @@ export const parse = (lines: string): EnvoyStats => {
 
     return prev
   }, {} as MetricRecord)
+
+  const { http, tcp, listener = {}, cluster = {} } = json as Protocols & {
+    listener: ConnectionStats
+    cluster: ConnectionStats
+  }
+
+  const { ...unknown } = cluster
+  return {
+    cluster: {
+      http,
+      tcp,
+      unknown,
+    },
+    listener,
+  }
 }
 
 // lets you specify the things you are interested in
-export const getTraffic = (json: EnvoyStats, filter: (key: string) => boolean): TrafficEntry[] => {
-  const protocols = ['http', 'tcp', 'cluster'] as const
+export const getTraffic = (json: ConnectionStats, filter: (key: string) => boolean): TrafficEntry[] => {
+  const _protocols = [...protocols, 'unknown'] as const
   const traffic: Record<string, TrafficEntry> = {}
-  protocols.map((item) => {
+
+  _protocols.map((item) => {
     return Object.entries(json[item] || {}).filter(([key, _value]) => {
       return filter(key)
     }).forEach(([key, value]) => {
       if (typeof traffic[key] === 'undefined') {
         traffic[key] = {
           name: key,
+          // FIXME Cope with _8888-sdofjsdf-big-hash
+          // we only get these on outbounds at the moment (and we don't need the port for the outbound)
+          // so we can pretend this doesn't happen
           port: key.split('_').at(-1) ?? '',
-          // TODO(jc): Not totally sure how we are going to get the protocol
-          // from the outbounds for gateways (these only have `cluster.`) so
-          // lets set it to `tcp` for now so as to now show `Cluster` as the
-          // protocol
-          protocol: item === 'cluster' ? 'tcp' : item,
+          // FIXME we need to detect the protocol based on the existence of
+          // a http property in the stats, just not sure which one yet
+          protocol: item === 'unknown' ? 'http' : item,
         }
-      }
-      // we only look at cluster to find the grpc stats
-      if (item === 'cluster') {
-        if (typeof value.grpc !== 'undefined') {
-          // move it from cluster.service-name.grpc.* to service-name.grpc.* if we find it
-          // the entry will already have a http protocol and a .http prop
-          // so change the protocol and add the grpc stats (additional to `.http` stats)
-          traffic[key].protocol = 'grpc'
-          traffic[key].grpc = value
+        traffic[key][traffic[key].protocol] = {
+          ...(traffic[key][traffic[key].protocol] || {}),
+          ...value as Record<string, any>,
         }
-        return
       }
-      traffic[key][item] = value
+      if (item === 'unknown') {
+        const obj = value as Record<string, any>
+        appProtocols.forEach((protocol) => {
+          if (typeof obj[protocol] !== 'undefined') {
+            traffic[key].protocol = protocol
+            traffic[key][protocol] = obj[protocol]
+          }
+        })
+      }
```

Comment:
This is the bit where we detect `grpc` or `http2` on `unknown` and hoist it up. Its the same approach as the red that was removed.